### PR TITLE
Refactor CsvReader constructor

### PR DESCRIPTION
**Pull request by chcn**

### CODE REVIEW

File: job/server/src/main/java/alluxio/job/transform/format/csv/CsvReader.java [hunk 1]
```diff
@@ -70,23 +86,10 @@ private static InputStream open(FileSystem fs, Path path) throws IOException {
...
-      String schemaName = inputPath.getName();
-      if (schemaName.contains(".")) {
-        schemaName = schemaName.substring(0, schemaName.indexOf("."));
-      }
-      schema = AvroCSV.inferSchema(schemaName, inputStream, props);
-    }
-    AvroCSVReader<Record> reader = new AvroCSVReader<>(open(fs, inputPath), props, schema,
-        Record.class, false);
-    return new CsvReader(fs, reader, schema);
+    return new CsvReader(fs, inputPath);
```
Comment (calvinjia) [inline job/server/src/main/java/alluxio/job/transform/format/csv/CsvReader.java]: Can we do `new CsvReader(inputPath)` and do the conf/fs creation in the constructor?

Comment (chcn) [reply job/server/src/main/java/alluxio/job/transform/format/csv/CsvReader.java]: sure

**Comment by chcn** [comment]
@calvinjia PTAL

File: job/server/src/main/java/alluxio/job/transform/format/csv/CsvReader.java [hunk 2]
```diff
@@ -46,27 +46,27 @@
...
-   * @param fs the hdfs compatible client
    * @param inputPath the input path
    * @throws IOException when failed to open the input
    */
-  private CsvReader(FileSystem fs, Path inputPath) throws IOException {
+  private CsvReader(Path inputPath) throws IOException {
     mCloser = Closer.create();
-    mFs = fs;
+    Configuration conf = ReadWriterUtils.readNoCacheConf();
+    mFs = inputPath.getFileSystem(conf);
```
Comment (calvinjia) [inline job/server/src/main/java/alluxio/job/transform/format/csv/CsvReader.java]: You can do `mFs = mCloser.register(inputPath.getFileSystem(conf))` (applies to `mReader` too)

File: job/server/src/main/java/alluxio/job/transform/format/csv/CsvReader.java [hunk 3]
```diff
@@ -46,27 +46,27 @@
...
+    try (InputStream inputStream = open(mFs, inputPath)) {
       String schemaName = inputPath.getName();
       if (schemaName.contains(".")) {
         schemaName = schemaName.substring(0, schemaName.indexOf("."));
       }
       schema = AvroCSV.inferSchema(schemaName, inputStream, props);
       mSchema = new CsvSchema(schema);
     }
-    mReader = new AvroCSVReader<>(open(fs, inputPath), props, schema, Record.class, false);
+    mReader = new AvroCSVReader<>(open(mFs, inputPath), props, schema, Record.class, false);
```
Comment (calvinjia) [inline job/server/src/main/java/alluxio/job/transform/format/csv/CsvReader.java]: If we throw an exception in the constructor should we make sure to close the closer?

**Comment by chcn** [comment]
@calvinjia PTAL

File: job/server/src/main/java/alluxio/job/transform/format/csv/CsvReader.java (hunk 2 above)
Comment (calvinjia) [reply job/server/src/main/java/alluxio/job/transform/format/csv/CsvReader.java]: @chcn you don't like this?

**Comment by chcn** [comment]
jenkins, test this please

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Comment by calvinjia** [comment]
alluxio-bot, cherry-pick this to branch-2.1 please.

**Comment by calvinjia** [comment]
alluxio-bot, cherry-pick this to branch-2.1 please.

**Pull request merged**